src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -187,7 +187,9 @@ private static void UpdateMetrics(ExecutionOptions opts, UInt256 effectiveGasPri
         {
             if (opts is ExecutionOptions.Commit or ExecutionOptions.None)
             {
-                float gasPrice = (float)((double)effectiveGasPrice / 1_000_000_000.0);
+                // log2(3e38) ~ 127.8, if effectiveGasPrice is more than 128bit long it may lead to an overflow
+                float gasPrice = effectiveGasPrice[2] != 0 || effectiveGasPrice[3] != 0 ? 3e29f : (float)((double)effectiveGasPrice / 1_000_000_000.0);
```

Comment:
```suggestion
                float gasPrice = effectiveGasPrice[2] != 0 || effectiveGasPrice[3] != 0 ? float.MaxValue : (float)((double)effectiveGasPrice / 1_000_000_000.0);
```

Reply:
I've just divided `effectiveGasPrice` by 1_000_000_000 as if it'd be `float.MaxValue` / 1e9. Code below potentially has this number multiplied by tx count, may be good to have some room. Or maybe I'm overthinking

Reply:
Makes sense. Could even go with a lower value or early exit if price is that high.

Is just for logging range boundaries; and an outlier gas price larger than anyone could pay for a single unit of not sure is helpful info? 🤔